tasks/nc_download.yml

```diff
@@ -1,6 +1,21 @@
 ---
 - name: "[NC-DL] - Unzip is installed"
   package: name=unzip state=installed
+  when: nextcloud_archive_format == "zip"
+
+- name: "[NC-DL] - bunzip2 is installed"
+  package: name=bzip2 state=installed
+  when: nextcloud_archive_format == "tar.bz2"
+
+- name: "[NC-DL] - regenerate the download URL for *latest*."
+  set_fact:
+    nextcloud_full_url: "{{nextcloud_repository}}/{{nextcloud_version_channel}}/{{[nextcloud_dl_file_name.latest, nextcloud_archive_format]|join('.')}}"
+  when: nextcloud_get_latest and (nextcloud_version_channel != 'prereleases') and (nextcloud_archive_url is not defined)
+
+- name: "[NC-DL] - use the provided URL instead."
```

Comment:
I'd move this one before the other one above and then check if `nextcloud_full_url` is set, to skip all the current `when` checks.

Also, why not only use `nextcloud_archive_url` **or** `nextcloud_full_url`? That would save 1 variable name and prevent any ambiguity/confusion between the two.